nimpy.nimble

```diff
@@ -18,20 +51,33 @@ proc runTests(nimFlags = "") =
     if sf.ext == ".nim" and not sf.name.startsWith("t"):
       exec "nim c --threads:on --app:lib " & nimFlags & " --out:" & f.path.changeFileExt(pluginExtension) & " " & f.path
 
-  mvFile("tests/custommodulename".changeFileExt(pluginExtension), "tests/_mycustommodulename".changeFileExt(pluginExtension))
+  let
+    sourceFile = "tests/custommodulename".changeFileExt(pluginExtension)
+    targetFile = "tests/_mycustommodulename".changeFileExt(pluginExtension)
+  try:
+    rmFile(targetFile)
+  except:
+    discard
+  mvFile(sourceFile, targetFile)
+
+  let
+    pythonExes = calcPythonExecutables()
+    libPythons = calcLibPythons()
 
   for f in oswalkdir.walkDir("tests"):
     # Run all python modules starting with "t"
     let sf = f.path.splitFile()
     if sf.ext == ".py" and sf.name.startsWith("t"):
-      exec "python2 " & f.path
-      exec "python3 " & f.path
+      for pythonExe in pythonExes:
+        echo "Testing Python executable: ", pythonExe
+        exec pythonExe & " " & f.path
 
   for f in oswalkdir.walkDir("tests"):
     # Run all nim modules starting with "t"
     let sf = f.path.splitFile()
     if sf.ext == ".nim" and sf.name.startsWith("t"):
-      exec "nim c -r " & nimFlags & " " & f.path
+      for libPython in libPythons:
+        exec "nim c -d:nimpytest -d:TEST_LIB_PYTHON=" & libPython & " -r " & nimFlags & " " & f.path
```

Comment:
`nimpytest` is redundant to `TEST_LIB_PYTHON` here. It should be just `-d:nimpyTestLibPython=...` and in code instead of `when defined(nimpy_test):` it should be `when nimpyTestLibPython.len != 0`...`

Reply:
I can take that out if you'd like.  The only reason I put it in there was to not leave any test related symbols in when people compile for regular use, but that probably would get optimized out when it is not used in regular code anyways though.